openldap/2.6/debian-11/rootfs/opt/bitnami/scripts/libopenldap.sh

```diff
@@ -0,0 +1,1045 @@
+        fi
+        debug_execute ldapadd "${slapd_debug_args[@]}" -Y EXTERNAL -H "$LDAP_LDAPI_URI" -f "${LDAP_CONF_DIR}/schema/${schema}.ldif"
+    done
+}
+
+########################
+# Add custom schema
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns
+#   None
+#########################
+ldap_add_custom_schema() {
+    info "Adding custom Schema from ${LDAP_CUSTOM_SCHEMA_FILE} ..."
+    debug_execute slapadd "${slapd_debug_args[@]}" -F "$LDAP_ONLINE_CONF_DIR" -n 0 -l  "$LDAP_CUSTOM_SCHEMA_FILE"
+    ldap_stop
+    while is_ldap_running; do sleep 1; done
+    ldap_start_bg
+}
+
+########################
+# Add custom schemas
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns
+#   None
+#########################
+ldap_add_custom_schemas() {
+    info "Adding custom schemas in ${LDAP_CUSTOM_SCHEMA_DIR} ..."
+    for schema in $(find "$LDAP_CUSTOM_SCHEMA_DIR" -maxdepth 1 \( -type f -o -type l \) -iname '*.ldif' -print | sort); do
+        info "\t${schema}"
+        debug_execute slapadd "${slapd_debug_args[@]}" -F "${LDAP_ONLINE_CONF_DIR}" -n 0 -l "${schema}"
+    done
+    ldap_stop
+    while is_ldap_running; do sleep 1; done
+    ldap_start_bg
+}
+
+########################
+# Create LDAP tree
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+ldap_create_tree() {
+    info "Creating LDAP default tree"
+    local dc=""
+    local o="example"
+    read -r -a root <<< "$(tr ',;' ' ' <<< "${LDAP_ROOT}")"
+    for attr in "${root[@]}"; do
+        if [[ $attr = dc=* ]] && [[ -z "$dc" ]]; then
+            dc="${attr:3}"
+        elif [[ $attr = o=* ]] && [[ $o = "example" ]]; then
+            o="${attr:2}"
+        fi
+    done
+    cat > "${LDAP_SHARE_DIR}/tree.ldif" << EOF
+# Root creation
+dn: $LDAP_ROOT
+objectClass: dcObject
+objectClass: organization
+dc: $dc
+o: $o
+
+dn: ${LDAP_USER_DC/#/ou=},${LDAP_ROOT}
+objectClass: organizationalUnit
+ou: users
+
+EOF
+    read -r -a users <<< "$(tr ',;' ' ' <<< "${LDAP_USERS}")"
+    read -r -a passwords <<< "$(tr ',;' ' ' <<< "${LDAP_PASSWORDS}")"
+    local index=0
+    for user in "${users[@]}"; do
+        cat >> "${LDAP_SHARE_DIR}/tree.ldif" << EOF
+# User $user creation
+dn: ${user/#/cn=},${LDAP_USER_DC/#/ou=},${LDAP_ROOT}
+cn: User$((index + 1 ))
+sn: Bar$((index + 1 ))
+objectClass: inetOrgPerson
+objectClass: posixAccount
+objectClass: shadowAccount
+userPassword: ${passwords[$index]}
+uid: $user
+uidNumber: $((index + 1000 ))
+gidNumber: $((index + 1000 ))
+homeDirectory: /home/${user}
+
+EOF
+        index=$((index + 1 ))
+    done
+    cat >> "${LDAP_SHARE_DIR}/tree.ldif" << EOF
+# Group creation
+dn: ${LDAP_GROUP/#/cn=},${LDAP_USER_DC/#/ou=},${LDAP_ROOT}
+cn: $LDAP_GROUP
+objectClass: groupOfNames
+# User group membership
+EOF
+
+    for user in "${users[@]}"; do
+        cat >> "${LDAP_SHARE_DIR}/tree.ldif" << EOF
+member: ${user/#/cn=},${LDAP_USER_DC/#/ou=},${LDAP_ROOT}
+EOF
```

Comment:
Best practice is to keep user and group containers separate (e.g. `ou=users, ...`,  `ou=groups, ...`)

Reply:
I think I've accomplished this, let me know if I didn't. :)

Reply:
I don't see how, have you tested this?

For example, I'd expect a user DN to look like:
```
cn=userid,ou=People,dc=example,dc=com
```

And a group DN:
```
cn=group1,ou=Groups,dc=example,dc=com
```

Reply:
Right, Github not always showing the most up to date diff.

Reply:
So, that means you're happy with this change?

Reply:
Yeah